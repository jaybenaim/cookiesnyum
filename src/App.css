html {
  position: relative;
  min-height: 100%;
}

body {
  margin-bottom: 140px !important; /* Height of the footer */
  padding: 0;
  overflow-x: hidden;
}

footer {
  position: absolute;
  bottom: 0;
  width: 76%;
  height: 140px; /* Set the fixed height of the footer here */
}

/* Override Bootstrap Styling */

.btn-primary {
  background-color: #025fce !important;
}

button:focus,
button:active {
  outline: none !important;
  box-shadow: none !important;
}

/* helpers */

.hide {
  display: none;
}
