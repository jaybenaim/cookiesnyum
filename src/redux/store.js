import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import rootReducer from "./reducers";
import { createLogger } from "redux-logger";

const loggerMiddleware = createLogger();
const initialState = {};
const middleware = [thunk, loggerMiddleware];

const store = createStore(
  rootReducer,
  initialState,
  applyMiddleware(...middleware)
);
export default store;
