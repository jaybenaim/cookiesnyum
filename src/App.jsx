import React, { useState, useEffect } from "react";
import { Switch, Route, withRouter } from "react-router-dom";
import "./App.css";
import {
  Footer,
  Register,
  Login,
  Home,
  Products,
  ProductShow,
  ErrorPage,
  CheckoutForm
} from "./components";
import { connect } from "react-redux";
import store from "./redux/store";
import jwt_decode from "jwt-decode";
import setAuthToken from "./utils/setAuthToken";
import { setCurrentUser, logoutUser } from "./redux/actions/authActions";
import Nav from "./components/Nav";
import Cart from "./components/Cart";

// Check for token to keep user logged in
if (localStorage.jwtToken) {
  // Set auth token header auth
  const token = localStorage.jwtToken;
  setAuthToken(token);
  // Decode token and get user info and exp
  const decoded = jwt_decode(token);
  // Set user and isAuthenticated
  store.dispatch(setCurrentUser(decoded));
  // Check for expired token
  const currentTime = Date.now() / 1000; // to get in milliseconds
  if (decoded.exp < currentTime) {
    // Logout user
    store.dispatch(logoutUser());
    // Redirect to login
    window.location.href = "./login";
  }
}

const App = ({ errors, history }) => {
  const [checkoutForm, showCheckoutForm] = useState(false);

  const handleCheckoutCart = data => {
    showCheckoutForm(!checkoutForm);
  };
  //TODO Web Template Studio: Adrsd routes for your new pages here.

  const cartOptions = {
    currencySymbol: "CAD",
    checkoutTextLabel: "Confirm",
    subTotalTextLabel: "Sub Total",
    cartTextLabel: "Cart",
    quantityTextLabel: "QTY"
  };

  useEffect(() => {
    errors && errors.isAxiosError && history.push("/404");
  }, [errors]);

  return (
    <React.Fragment>
      {errors.length >= 1 ? (
        ""
      ) : (
        <>
          <Nav />
          <Cart
            {...cartOptions}
            handleCheckout={handleCheckoutCart}
            checkoutForm={checkoutForm}
          />
        </>
      )}
      <Switch>
        <>
          <Route exact path="/" render={props => <Home {...props} />} />
          <Route
            exact
            path="/register"
            render={props => <Register {...props} />}
          />
          <Route exact path="/login" render={props => <Login {...props} />} />
          <Route
            exact
            path="/products"
            render={props => <Products {...props} />}
          />
          <Route
            exact
            path="/products/:name"
            render={props => <ProductShow {...props} />}
          />
          <Route
            exact
            path="/checkout"
            render={props => <CheckoutForm {...props} />}
          />
          <Route exact path="/404" render={props => <ErrorPage {...props} />} />
        </>
      </Switch>
      <Footer />
    </React.Fragment>
  );
};

const mapStateToProps = state => ({
  auth: state.auth,
  errors: state.errors
});
export default withRouter(connect(mapStateToProps, {})(App));
