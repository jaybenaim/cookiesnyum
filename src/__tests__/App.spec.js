import React from "react";
import { Provider } from "react-redux";
import store from "../redux/store";
import App from "../App";
import renderer from "react-test-renderer";

import { HashRouter } from "react-router-dom";
import ErrorBoundary from "../components/ErrorBoundary";

describe("App", () => {
  test("App renders", () => {
    console.log = jest.fn();
    const component = renderer.create(
      <Provider store={store}>
        <HashRouter basename="/dolcenada">
          <ErrorBoundary>
            <App />
          </ErrorBoundary>
        </HashRouter>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(console.log).toHaveBeenCalled();
  });
});
