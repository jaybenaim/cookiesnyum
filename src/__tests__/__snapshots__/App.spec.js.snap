// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App renders 1`] = `
<div
  id="notfound"
>
  <div
    class="notfound"
  >
    <div
      class="notfound-404"
    >
      <h1>
        404
      </h1>
    </div>
    <h2>
      Oops, The Page you are looking for can't be found!
    </h2>
    <a
      href="#/dolcenada/"
      onClick={[Function]}
    >
      <span
        class="arrow"
      />
      Return To Homepage
    </a>
  </div>
</div>
`;
